client/src/components/ScoreForm.js

```diff
@@ -59,19 +107,28 @@ const TableBonuses = ({ bonuses, handleCheckedBox }) => (
 			</Table.Row>
 		</Table.Header>
 		<Table.Body>
-			{bonuses.map(bonus => {
-				const { mo,tu,we,th,fr,sa,su } = {...bonus}
+			{bonuses.map((bonus, index) => {
+				const name = 'user.bonuses[' + index + '].value.'
+				const fieldName = {
+					'mo': name + 'mo',
+					'tu': name + 'tu',
+					'we': name + 'we',
+					'th': name + 'th',
+					'fr': name + 'fr',
+					'sa': name + 'sa',
+					'su':	name + 'su'
+				}
 				return (
 					<Table.Row key={bonus.name}>
 						<Table.Cell width={3}>{bonus.name}</Table.Cell>
-						<Table.Cell>	
-							<Checkbox label='Mon' checked={mo} onChange={handleCheckedBox}/>
-							<Checkbox label='Tues' checked={tu}/>
-							<Checkbox label='Wed' checked={we}/>
-							<Checkbox label='Thurs' checked={th}/>
-							<Checkbox label='Fri' checked={fr}/>
-							<Checkbox label='Sat' checked={sa}/>
-							<Checkbox label='Sun' checked={su}/>
+						<Table.Cell>
+							<Field name={fieldName.mo} component={Bonus} type='checkbox' label='Mon' />	
+							<Field name={fieldName.tu} component={Bonus} type='checkbox' label='Tue' />	
+							<Field name={fieldName.we} component={Bonus} type='checkbox' label='Wed' />	
+							<Field name={fieldName.th} component={Bonus} type='checkbox' label='Thurs' />	
+							<Field name={fieldName.fr} component={Bonus} type='checkbox' label='Fri' />	
```

Comment:
fieldName.mo will check the fieldName object for the mo property and use that property value as the name for the Field component. For fieldName.mo, this would be user.bonuses[index].value.mo. This is necessary as redux-form matches the name prop with the initialized data.

+const initialState = {
 +	user: {
 +		exercises: data.exercises,
 +		bonuses: data.bonuses
 +	}

which currently gets its data from scoreForm-fakedata.json.